@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -418,33 +459,102 @@ export class TextElementsRenderer {
         this.m_options.showReplacementGlyphs = value;
 
         this.m_textCanvases.forEach(textCanvas => {
-            textCanvas.fontCatalog.showReplacementGlyphs = value;
+            if (textCanvas?.fontCatalog) {
+                textCanvas.fontCatalog.showReplacementGlyphs = value;
+            }
         });
     }
 
+    restoreRenderers(renderer: THREE.WebGLRenderer) {
+        this.m_renderer = renderer;
+        this.m_poiRenderer = new PoiRenderer(
+            this.m_renderer,
+            this.m_poiManager,
+            this.m_imageCaches
+        );
+        //TODO: restore TextCanvasRenderers
+    }
+
     /**
-     * Render the text using the specified camera into the current canvas.
+     * Updates the FontCatalogs used by this {@link TextElementsRenderer}.
      *
-     * @param camera - Orthographic camera to use.
+     * @param fontCatalogs - The new list of {@link FontCatalogConfig}s
      */
-    renderText(camera: THREE.OrthographicCamera) {
-        if (!this.initialized) {
-            return;
+    async updateFontCatalogs(fontCatalogs?: FontCatalogConfig[]) {
+        if (this.m_defaultFontCatalogConfig) {
+            if (
+                !fontCatalogs ||
+                fontCatalogs.findIndex(config => {
+                    return config.name === DEFAULT_FONT_CATALOG_NAME;
+                }) === -1
+            ) {
+                // not other default catalog available, keep the old one
+                if (!fontCatalogs) {
+                    fontCatalogs = [];
+                }
+                // Never remove the default Canvas if set per configuration
+                fontCatalogs?.unshift(this.m_defaultFontCatalogConfig);
+            } else {
+                if (this.m_textCanvases.has(DEFAULT_FONT_CATALOG_NAME)) {
+                    this.m_textCanvases.delete(DEFAULT_FONT_CATALOG_NAME);
+                }
+            }
+        }
+
+        if (fontCatalogs && fontCatalogs.length > 0) {
+            // Remove obsolete ones
+            for (const [name] of this.m_textCanvases) {
+                if (
+                    fontCatalogs.findIndex(catalog => {
+                        return catalog.name === name;
+                    }) < 0
+                ) {
+                    this.m_textCanvases.delete(name);
+                }
+            }
+
+            // Add new catalogs
+            for (const fontCatalog of fontCatalogs) {
+                await this.addTextCanvas(fontCatalog);
+            }
+        } else {
+            this.m_textCanvases.clear();
         }
+        this.m_textStyleCache.initializeTextElementStyles(this.m_textCanvases);
```

Comment:
Is this needed both here and in updateTextStyles? Because MapView calls updateFontCatalogs and updateTextStyles immediately after.

Reply:
currently this is still needed, as in theory both methods can be called separately,.. 
But, I think in the next step in this process, which will make the TextStyleCache updatable instead of recreating a new one every time the theme is set, this will most probably disappear.